source/code/troubleshooter/omsagent_tsg_zip/install_tsg

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# check if running as root
+if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root in order to install the troubleshooter, due to the permissions required for the destination directories."
+    echo "Please instead run:"
+    echo ""
+    echo "$ sudo ./install_tsg"
+    echo ""
+    echo "in order to install the troubleshooter."
+    exit 1
+fi
+
+bin_path="/opt/microsoft/omsagent/bin/"
+omsagent_tsg_path="${bin_path}/omsagent_tsg"
+plugin_tsg_path="/opt/microsoft/omsagent/plugin/troubleshooter"
+
+echo "Getting ready to install troubleshooter..."
+
+# set up machine
+
+if [ -d $plugin_tsg_path ]; then
+    # get rid of old files to update
+    echo "Removing older version of troubleshooter..."
+    rm -rf $plugin_tsg_path
+    rm -f $omsagent_tsg_path
+fi
+
+# copy over files
+echo "Installing troubleshooter on machine..."
+mkdir $plugin_tsg_path
```

Comment:
Changed all instances of `mkdir` to `mkdir -p` just in case:
https://github.com/microsoft/OMS-Agent-for-Linux/blob/bff7c80f869d6f5a4046ed41b4bd0ed89d69219f/source/code/troubleshooter/install_tst#L32